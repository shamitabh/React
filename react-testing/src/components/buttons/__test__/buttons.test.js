import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Buttons from "../buttons";
import TestRenderer from "react-test-renderer";

// remove the divs created as part of the test cases after each test
afterEach(cleanup);

test("renders without crashing - appraoch 1", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Buttons />, div);
});

test("renders without crashing - appraoch 2", () => {
  render(<Buttons />);
});

test("renders Button correctly", () => {
  // getByTextId method comes from the use of dat-testid in the divs within Button component
  const { getByTestId } = render(<Buttons label="save" />);
  expect(getByTestId("buttons")).toHaveTextContent("save");
});

// refer to the youtube video for snapshot
test("matches snapshot", () => {
  // toJSON creates a javascript object for the HTML elements rendered
  const tree = TestRenderer.create(<Buttons />).toJSON();
  expect(tree).toMatchSnapshot();
});
