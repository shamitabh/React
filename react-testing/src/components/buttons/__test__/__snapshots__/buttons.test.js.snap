// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 1`] = `
<div
  className="button-style"
  data-testid="buttons"
/>
`;

exports[`matches snapshot 1`] = `
<div
  className="button-style"
  data-testid="buttons"
/>
`;
